app/src/main/java/ch/hikemate/app/model/route/HikesViewModel.kt

```diff
@@ -296,15 +327,7 @@ class HikesViewModel(
    *   operation.
    */
   fun retrieveLoadedHikesOsmData(onSuccess: () -> Unit = {}, onFailure: () -> Unit = {}) =
-      viewModelScope.launch {
-        // Notify the UI that a heavy loading operation is in progress
-        _loading.value = true
-
-        retrieveLoadedHikesOsmDataAsync(onSuccess, onFailure)
-
-        // The heavy loading operation is done now
-        _loading.value = false
-      }
+      viewModelScope.launch { retrieveLoadedHikesOsmDataAsync(onSuccess, onFailure) }
```

Comment:
Question: could you explain why `setLoading()` is not called here?

Reply:
Good point, I moved it to inside of `retrieveLoadedHikesOsmDataAsync` directly. This is because I wanted to set `loading` to `true` if and only if a loading operation was actually performed. The method first does some checks about whether the loading operation is necessary, and performs an early return if the request is not needed.

So I only call `setLoading` when the request is actually being performed.

Reply:
I understand why you did it, but it seems to me that that is inconsistent with the rest of the code since I see that you always call the `setLoading` function in the wrapper functions and not in the `async` functions.
But if there isn't any way to solve this I think you can leave it as is, since the code is robust enough anyways since it handles the case where there is an error well: taking into account the `setLoading(false)` call is outside the `withLock` call, so anyways the code would be equivalent.

Reply:
I think we can leave it like this, it doesn't pose any real problems.